python/paddle/v2/framework/graph.py

```diff
@@ -1,50 +1,121 @@
 import paddle.v2.framework.core as core
 import collections
+import numpy as np
+import copy
 
 __all__ = ['Block', 'Variable', 'Program', 'Operator']
 
 
 class Variable(object):
-    def __init__(self, block, name=None, shape=None, dtype=None,
-                 lod_level=None):
+    def __init__(self,
+                 block,
+                 name=None,
+                 shape=None,
+                 dtype=None,
+                 lod_level=None,
+                 **kwargs):
         self.block = block
 
         if name is None:
             name = Variable._unique_var_name_()
-        self.proto = self.block.proto.new_var(name)
+        try:
+            self.desc = self.block.desc.var(name)
+            is_new_var = False
```

Comment:
When will the constructor be called for an existing var?

Reply:
Two layers share a same parameter.